cmd/config-shard-validator/main.go

```diff
@@ -118,3 +138,25 @@ func main() {
 		logrus.Fatal("Found configurations that do not belong to the correct auto-updating config")
 	}
 }
+
+func checkSpec(spec *v1.PodSpec, info *jobconfig.Info, relPath, name string) {
+	for i, container := range spec.Containers {
+		for _, env := range container.Env {
+			if env.Name == "CONFIG_SPEC" && env.ValueFrom != nil && env.ValueFrom.ConfigMapKeyRef != nil {
+				configInfo, err := config.InfoFromPath(path.Join(info.Org, info.Repo, env.ValueFrom.ConfigMapKeyRef.Key))
```

Comment:
This is fragile since it assumes that someone is configuring a job for $org/$repo with the matching ci-operator config for $org/$repo ... but this presubmit will fail if they don't honor that and that seems fine to me. This bit us on periodics where we could not grab org/repo ownership from prow config since periodics are unrelated